test/testHeadless.m

```diff
@@ -35,6 +36,7 @@
         makeTravisReport(status);
     end
 
+    restoreGlobalState(state);
```

Comment:
Wasn't the `try-finally` construct not available in matlab, but only `try-catch`? 
This is why the PR is labeled WIP -- I am open to suggestions and gladly follow suggestions. :-)